bin/git-linguist

```diff
@@ -118,27 +118,33 @@ def git_linguist(args)
 
   parser.parse!(args)
 
-  git_dir = `git rev-parse --git-dir`.strip
-  raise "git-linguist must be run in a Git repository (#{Dir.pwd})" unless $?.success?
-  wrapper = GitLinguist.new(git_dir, commit, incremental)
-
-  case args.pop
-  when "stats"
-    wrapper.linguist do |linguist|
-      puts JSON.dump(linguist.languages)
-    end
-  when "breakdown"
-    wrapper.linguist do |linguist|
-      puts JSON.dump(linguist.breakdown_by_file)
+  begin
+    git_dir = `git rev-parse --git-dir`.strip
+    raise "git-linguist must be run in a Git repository" unless $?.success?
+    wrapper = GitLinguist.new(git_dir, commit, incremental)
+
+    case args.pop
+    when "stats"
+      wrapper.linguist do |linguist|
+        puts JSON.dump(linguist.languages)
+      end
+    when "breakdown"
+      wrapper.linguist do |linguist|
+        puts JSON.dump(linguist.breakdown_by_file)
+      end
+    when "dump-cache"
+      puts JSON.dump(wrapper.load_language_stats)
+    when "clear"
+      wrapper.clear_language_stats
+    when "disable"
+      wrapper.disable_language_stats
+    else
+      $stderr.print(parser.help)
+      exit 1
     end
-  when "dump-cache"
-    puts JSON.dump(wrapper.load_language_stats)
-  when "clear"
-    wrapper.clear_language_stats
-  when "disable"
-    wrapper.disable_language_stats
-  else
-    $stderr.print(parser.help)
+  rescue Exception => e
+    $stderr.puts e.message
+    $stderr.puts e.backtrace
     exit 1
```

Comment:
Something like this @carlosmn?

Reply:
Yeah. I would have just put the `rescue` a the bottom since a method implicitly starts a `begin` but this works just as well.

Reply:
Doh! Of course. I like that idea better. Coming up.

Reply:
Done!